plugins/ssr/ssripats.ml

```diff
@@ -48,39 +64,91 @@ type istate = {
   (* Delayed clear *)
   to_clear : Id.t list;
 
+  (* Temporary intros, to be generalized back *)
+  to_generalize : delayed_gen list;
+
+  (* The type of the inductive constructor corresponding to the current proof
+   * branch: name seeds are taken from that in an intro block *)
+  name_seed : EConstr.types option;
+
+}
+and delayed_gen = {
+  tmp_id : Id.t;    (* Temporary name *)
+  orig_name : Name.t   (* Old name *)
 }
 
 let empty_state = {
   to_clear = [];
+  to_generalize = [];
+  name_seed = None;
 }
 
 include Ssrcommon.MakeState(struct
   type state = istate
   let init = empty_state
 end)
 
+let ns_print env sigma = function
```

Comment:
What's `ns`? Can you expand the name?

Reply:
Oh wait, are all these debug functions? If yes, put them in a `Debug` module or whatever. Or at least delimit the debugging part by comments.